src/RSocketClient.h

```diff
@@ -56,13 +60,26 @@ class RSocketClient {
       std::unique_ptr<DuplexConnection> connection,
       folly::EventBase& eventBase,
       SetupParameters setupParameters = SetupParameters(),
-      std::shared_ptr<RSocketResponder> responder = std::shared_ptr<RSocketResponder>(),
-      std::unique_ptr<KeepaliveTimer> keepaliveTimer = std::unique_ptr<KeepaliveTimer>(),
+      std::shared_ptr<RSocketResponder> responder =
+          std::shared_ptr<RSocketResponder>(),
+      std::unique_ptr<KeepaliveTimer> keepaliveTimer =
+          std::unique_ptr<KeepaliveTimer>(),
       std::shared_ptr<RSocketStats> stats = std::shared_ptr<RSocketStats>(),
-      std::shared_ptr<RSocketNetworkStats> networkStats = std::shared_ptr<RSocketNetworkStats>());
+      std::shared_ptr<RSocketNetworkStats> networkStats =
+          std::shared_ptr<RSocketNetworkStats>());
+
+  folly::Future<folly::Unit> resume(
+      std::unique_ptr<ClientResumeStatusCallback> resumeCallback =
+          std::make_unique<DefaultResumeCallback>());
+
+  void disconnect(folly::exception_wrapper ex);
 
  private:
   std::unique_ptr<ConnectionFactory> connectionFactory_;
   std::unique_ptr<RSocketConnectionManager> connectionManager_;
+
+  std::shared_ptr<RSocketStateMachine> stateMachine_;
```

Comment:
The benefit of the way it's currently working, prior to this change is that a user only needs to hold the reference to the `RSocketRequester`, they don't need to hold both the `RSocketClient` and `RSocketRequester`.